app/src/main/java/org/projectbuendia/client/ui/SnackBar.java

```diff
@@ -0,0 +1,436 @@
+     * @param key The message key.
+     * @param seconds Seconds until dismiss.
+     */
+    private void setTimer(final MessageKey key, int seconds) {
+        if(seconds > 0) {
+            int limit = seconds*1000;
+            new CountDownTimer(limit, 1000) {
+                @Override public void onTick(long millisUntilFinished) {
+                }
+
+                @Override public void onFinish() {
+                    mMessagesList.remove(key);
+                    updateList();
+                }
+            }.start();
+        }
+    }
+
+    /**
+     * Programmatically dismiss a message by it's id.
+     * @param id The message id.
+     */
+    public void dismiss(int id) {
+        dismiss(getKey(id));
+    }
+
+    /**
+     * Programmatically dismiss a message by a MessageKey Object
+     * @param key The message Key.
+     */
+    public void dismiss(MessageKey key) {
+        mMessagesList.remove(key);
+        updateList();
+    }
+
+    /**
+     * Find message Key by it's id value.
+     * @param id The id of the message.
+     * @return The MessageKey of the message.
+     */
+    private MessageKey getKey(int id) {
+        MessageKey theKey = null;
+        for(Map.Entry<MessageKey, Message> entry : mMessagesList.entrySet()) {
+            MessageKey key = entry.getKey();
+            if (key.id == id) {
+                theKey = key;
+                break;
+            }
+        }
+        return theKey;
+    }
+
+    /**
+     * Rearrange the message data and notify the change to the adapter so it can redraw the
+     * ExpandableListView.
+     */
+    private void updateList() {
+        groups.clear();
+        children.clear();
+        children.add(new ArrayList<HashMap<String, String>>());
+        for(Map.Entry<MessageKey, Message> entry : mMessagesList.entrySet()) {
+            Message value = entry.getValue();
+
+            if (groups.size() == 0) {
+                addToHashMap(value, groups);
+            } else {
+                addToHashMap(value, children.get(0));
+            }
+        }
+        adapter.notifyDataSetChanged();
+    }
+
+    /**
+     * The main message list {@code mMessagesList} values are added to ArrayLists to be used by the
+     * SnackBarListAdapter.
+     * @param m The message to be stored.
+     * @param a The group or children ArrayList.
+     */
+    private void addToHashMap(Message m, ArrayList<HashMap<String, String>> a) {
+        HashMap<String, String> newMessage = new HashMap<>();
+        newMessage.put("id", Integer.toString(m.key.id));
+        newMessage.put("priority", Integer.toString(m.key.priority));
+        newMessage.put("messages", m.message);
+        if (m.actionString != null) {
+            newMessage.put("actions", m.actionString);
+        }
+        a.add(newMessage);
+    }
+
+    /**
+     * Initiates the SnackBar ExpandableListView.
+     */
+    private void buildList() {
+        mList = new ExpandableListView(mContext);
```

Comment:
I advise inflating the `ExpandableListView` from a layout so you can apply the styling there, instead of hard-coding style values into the `setListAppearance` method. Then, you could get rid of the `Utils.getPixelFromDips()` function too.